walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -20,118 +6,45 @@
 import java.util.List;
 import java.util.Set;
 
+/*
+  Assumptions: the collection of events is in sorted order by start time
+  If this assumption is not valid, then I would start by sorting the 
+  list of events (would increase runtime).
+*/
 public final class FindMeetingQuery {
 
   private static final int UPPER_BOUND = 1440;
   private static final int LOWER_BOUND = 0;
 
-  private List<TimeRange> getTimesThatDontWork(Collection<Event> events, Collection<String> attendees) {
-    List<TimeRange> blockedTimes = new ArrayList<>();
-
-    for (Event event : events) {
-      TimeRange eventTime = event.getWhen();
-      Set<String> peopleAtEvent = event.getAttendees();
-      for (String attendee : attendees) {
-        if (peopleAtEvent.contains(attendee)) {
-          blockedTimes.add(eventTime);
-        }
-      }
-    }
-
-    return blockedTimes;
-  } 
-
-  private List<TimeRange> condenseTimeRanges(List<TimeRange> blockedTimes) {
-    ArrayList<TimeRange> condensedBlockedTimes = new ArrayList<>();
-    for (int i = 0; i < blockedTimes.size(); i++) {
-      TimeRange first = blockedTimes.get(i);
-
-      if (i == blockedTimes.size() - 1) {
-        condensedBlockedTimes.add(first);
-        break;
-      }
-      TimeRange second = blockedTimes.get(i + 1);
-
-      if (first.overlaps(second)) { //can condense
-        int firstEnd = first.end();
-        int secondEnd = second.end();
-        int newStart = first.start();
-        
-        int newEnd = secondEnd;
-        if (firstEnd > secondEnd) {
-          newEnd = firstEnd;
-        } 
-
-        i++;
-        condensedBlockedTimes.add(TimeRange.fromStartDuration(newStart, newEnd - newStart));
-      } else {
-        condensedBlockedTimes.add(first);
-      }
-    }
-    return condensedBlockedTimes;
-  }
-
-  public Collection<TimeRange> otherway(Collection<Event> events, MeetingRequest request) {
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     Collection<String> attendees = request.getAttendees();
```

Comment:
nit: it's hard to differentiate attendees from allAttendees, I would name this one "requiredAttendees"

Reply:
Named it mandatoryAttendees - done